docs/README.rst

```diff
@@ -1046,6 +1046,71 @@ functionality for particular interfaces.
    how to override functions in interface definitions and why, prior
    to Python 3.6, the zero-argument version of `super` cannot be used.
 
+
+Persistency and Equality
+========================
+
+An important design goal has been persistency support for interfaces.
+This allows different processes to use the same interface and
+to create persistent associations between (persistent) objects
+and interfaces. For example, an application can store an object
+together with its provided interfaces in a database; later, potentially
+in a different invocation, it can search the database for objects
+providing a given interface.
+
+To make an object persistent, it must have a persistent identifier,
+PID for short.
+It is this PID which identifies the object across different
+processes. In the context of interfaces, we want to support
+evolution similarly to the evolution of classes: even though
+we change the interface (e.g. add a method, change documentation), we
+still want to consider it as the same interface.
+
+Python's main persistency support comes from its ``pickle`` module.
+It uses as PID for code objects (such as classes and functions)
+the combinations of their ``__module__`` and ``name`` attributes.
+In analogy, the PID for an interface is defined
+as the combination of its
+``__module__`` and ``__name__`` giving persistency behavior similar
+to classes, including evolution support.
+
+Unlike classes, interfaces define their
```

Comment:
I would probably also add a comment about how, *like* classes, only one object with a given `__module__` and `__name__` can be pickled at a time. 

```pycon
>>> class Foo:
...     pass
>>> f1 = Foo
>>> class Foo:
...     pass
>>> pickle.dumps(f1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
PicklingError: Can't pickle <class '__main__.Foo'>: it's not the same object as __main__.Foo
```

 If you do this dynamically without watching the assigned names, none can be pickled.
```pycon
>>> f1 = type('Foo', (), {})
>>> f2 = type('Foo', (), {})
>>> pickle.dumps(f1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
PicklingError: Can't pickle <class '__main__.Foo'>: it's not the same object as __main__.Foo
>>> del Foo
>>> pickle.dumps(f2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
PicklingError: Can't pickle <class '__main__.Foo'>: attribute lookup Foo on __main__ failed
```

All of the above holds for interfaces.

Classes, however, are inconsistent, or at least inconvenient: They compare and hash by identity (and can't be sorted) but pickle by global identifier, meaning that the value of `==` can change. 
```pycon
>>> class Foo:
...     pass
>>> f1 = Foo
>>> pickled_f1 = pickle.dumps(Foo)
>>> class Foo:
...     pass
>>> f1 == Foo
False
>>> unpickled_f1 = pickle.loads(pickled_f1)
>>> unpickled_f1
<class '__main__.Foo'>
>>> unpickled_f1 == Foo # Surprise!
True
>>> unpickled_f1 == f1
False
```

This is not the case for interfaces:
```pycon
>>> class IFoo(Interface):
...     pass
>>> f1 = IFoo
>>> pickled_f1 = pickle.dumps(f1)
>>> class IFoo(Interface):
...     pass
>>> IFoo == f1
True
>>> unpickled_f1 = pickle.loads(pickled_f1)
>>> unpickled_f1
<InterfaceClass __main__.IFoo>
>>> unpickled_f1 == IFoo == f1
True
```